File: java/jeromq/app/src/main/java/com/example/jeromq/ZeroMQMessageTask.java
package com.example.jeromq;

import android.os.AsyncTask;
import android.os.Handler;

import org.zeromq.ZMQ;

public class ZeroMQMessageTask extends AsyncTask<String, Void, String> {

	public ZeroMQMessageTask(Handler uiThread) {
		_uiThread = uiThread;
	}

	@Override
	protected String doInBackground(String ... params) {
		ZMQ.Context ctx = ZMQ.context(1);
		ZMQ.Socket req = ctx.socket(ZMQ.REQ);
		req.connect("tcp://127.0.0.1:5555");

		req.send(params[0].getBytes(), 0);
		String result = new String(req.recv(0));

		req.close();
		ctx.term();

		return result;
	}

	@Override
	protected void onPostExecute(String result) {
		_uiThread.sendMessage(Util.bundledMessage(_uiThread, result));
	}

	private final Handler _uiThread;
}

File: kotlin/jeromq/app/src/main/java/example/kotlin/jeromq/ZeroMQMessageTask.kt
package example.kotlin.jeromq

import android.os.AsyncTask
import android.os.Handler
import org.zeromq.ZMQ
import org.zeromq.ZContext

class ZeroMQMessageTask(private val uiThread: Handler) : AsyncTask<String, Void, String>() {

	override fun doInBackground(vararg params: String?): String {
		val ctx = ZContext(1)
		val req = ctx.createSocket(ZMQ.REQ)
		req.connect("tcp://127.0.0.1:5555")

		req.send(params[0]!!.toByteArray(), 0)
		val result = String(req.recv(0))

		req.close()
		ctx.close()

		return result
	}

	override fun onPostExecute(result: String?) {
		super.onPostExecute(result)
		if (result != null)
			uiThread.sendMessage(Util.bundledMessage(uiThread, result))
	}
}